File: algorithm/implement/prb3 게임 개발/mine2.c
import time

n,m = map(int,input().split())
x,y,dir = map(int,input().split())

map_data = []

for i in range(n):
  map_data.append(list(map(int, input().split())))

start_time = time.time()
flg_back = 0
result = 1

dx = [-1,0,+1,0]
dy = [0,+1,0,-1]

def turn_left():
  global dir
  if dir == 0:
      dir=3
  else : dir = dir - 1

while(1):
  #발자국 찍기
  map_data[x][y] = 2
  
  for i in range(4):
    #방향 돌기
    turn_left()

    #갈수있나 확인
    if map_data[x + dx[dir]][y + dy[dir]] == 0:
      x = x + dx[dir]
      y = y + dy[dir]
      result += 1
      break

    if i == 3:
      print("no place to move")
      flg_back = 1

      x = x - dx[dir]
      y = y - dy[dir]
      
      if map_data[x][y] == 1:
        break

      turn_left()
  if flg_back == 1:
    break

print(map_data)
print(result)
  
end_time = time.time()

print(end_time - start_time)

File: algorithm/Dynamic Programming/Q33 퇴사/mine2.py
# n = int(input())
n=10
# graph = [[] for _ in range(n+1)]
t = [-1] * (n+1)
p = [-1] * (n+1)
db = [-1] * (n+1)
maxval = -1

# for i in range(1,n+1):
#   t,p = map(int,input().split())
#   graph[i].append((t,p))

graph = [[],[1,1],[1,2],[1,3],[1,4],[1,5],[1,6],[1,7],[1,8],[1,9],[1,10]]

for i in range(1,n+1):
  end = i+graph[i][0]-1
  price = graph[i][1]
  if end < n+1:
    t[i] = end
    p[i] = price
  else:
    p[i] = 0

for i in range(n,0,-1):
  if t[i] == [-1]:
    db[i] = (0)
  else:
    if t[i]+1 <= n:
      plus = db[t[i]+1]
    else:
      plus = 0
    db[i] = (p[i] + plus)
    if maxval < db[i]:
      maxval = db[i]

print(maxval)
